Orange/widgets/utils/listfilter.py

```diff
@@ -0,0 +1,236 @@
+    Key up/ down to navigate between posible completions.
+    """
+    def eventFilter(self, obj, event):
+        if (event.type() == QEvent.KeyPress and
+                isinstance(obj, QLineEdit)):
+            if event.key() == Qt.Key_Down:
+                diff = 1
+            elif event.key() == Qt.Key_Up:
+                diff = -1
+            else:
+                return False
+            completer = obj.completer()
+            if completer is not None and completer.completionCount() > 0:
+                current = completer.currentRow()
+                current += diff
+                completer.setCurrentRow(current % completer.completionCount())
+                completer.complete()
+            return True
+        else:
+            return False
+
+
+def variables_filter(parent=None, model=None):
+    def update_completer_model():
+        """ This gets called when the model for available attributes changes
+        through either drag/drop or the left/right button actions.
+
+        """
+        nonlocal original_completer_items
+        vars = list(model)
+        items = [var.name for var in vars]
+        items += ["%s=%s" % item for v in vars for item in v.attributes.items()]
+
+        new = sorted(set(items))
+        if new != original_completer_items:
+            original_completer_items = new
+            completer_model.setStringList(original_completer_items)
+
+    def update_completer_prefix():
+        """ Prefixes all items in the completer model with the current
+        already done completion to enable the completion of multiple keywords.
+        """
+        nonlocal original_completer_items
+        prefix = str(completer.completionPrefix())
+        if not prefix.endswith(" ") and " " in prefix:
+            prefix, _ = prefix.rsplit(" ", 1)
+            items = [prefix + " " + item
+                     for item in original_completer_items]
+        else:
+            items = original_completer_items
+        old = list(map(str, completer_model.stringList()))
+
+        if set(old) != set(items):
+            completer_model.setStringList(items)
+
+    original_completer_items = []
+
+    filter_edit = QLineEdit()
+    filter_edit.setToolTip("Filter the list of available variables.")
+    if hasattr(filter_edit, "setPlaceholderText"):
+        filter_edit.setPlaceholderText("Filter")
+
+    completer = QCompleter()
+    completer.setCompletionMode(QCompleter.InlineCompletion)
+    completer_model = QStringListModel()
+    completer.setModel(completer_model)
+    completer.setModelSorting(
+        QCompleter.CaseSensitivelySortedModel)
+
+    filter_edit.setCompleter(completer)
+    completer_navigator = CompleterNavigator(parent)
+    filter_edit.installEventFilter(completer_navigator)
+
+    proxy = VariableFilterProxyModel()
+    proxy.setSourceModel(model)
+    view = VariablesListItemView(
+        acceptedType=Orange.data.Variable)
+    view.setModel(proxy)
+
+    aa = model
```

Comment:
Why `aa`? Now that the `model` is not buried within some object, just use `model` instead of `aa`.